src/Core/src/Credentials/Credentials.php

```diff
@@ -21,16 +23,20 @@ final class Credentials implements CredentialProvider
 
     private $expireDate;
 
+    private $cacheable;
+
     public function __construct(
         string $accessKeyId,
         string $secretKey,
         ?string $sessionToken = null,
-        ?\DateTimeImmutable $expireDate = null
+        ?\DateTimeImmutable $expireDate = null,
+        bool $cacheable = false
```

Comment:
The purpose of this parameter is to tell if the Credential can be persisted.
Actually it's used once: When credential are fetched from .ini file.

I use it to avoid the following case:
- user call an API
- credential are fetched from .ini and stored in cache
- user update the .ini file
- user call the API
-> cached crednetial are re-used => user don't understand what happens


Do you think it worse it?

Reply:
Yes. I like it. Good thinking.